python/cudf/dataframe/numerical.py

```diff
@@ -158,17 +159,7 @@ def astype(self, dtype):
 
     def sort_by_values(self, ascending=True, na_position="last"):
         sort_inds = get_sorted_inds(self, ascending, na_position)
-        col_keys = cudautils.gather(data=self.data.mem,
-                                    index=sort_inds.data.mem)
-        mask = None
-        if self.mask:
-            mask = self._get_mask_as_column()\
-                .take(sort_inds.data.to_gpu_array()).as_mask()
-            mask = Buffer(mask)
-        col_keys = self.replace(data=Buffer(col_keys),
-                                mask=mask,
-                                null_count=self.null_count,
-                                dtype=self.dtype)
+        col_keys = cpp_copying.apply_gather_column(self, sort_inds.data.mem)
```

Comment:
Why is a gather needed to sort an array by value? I think this is only needed because they use get_sorted_inds rather than just sorting the values...  Why not just use libcudf order_by?

Reply:
I'm sorry to ask questions out of the scope of this PR...

Reply:
Under the hood we are using the libcudf `order_by` implementation, but I believe we just get the indices back from it and perform the gather from those indices. This is at least partially by design, because for the dataframe level `sort_values` function we need to gather the non-sorted columns based on the new ordering as well.